tests/integration/test_task_pool.py

```diff
@@ -1452,21 +1733,19 @@ def max_cycle(tasks):
         return max([int(t.tokens.get("cycle")) for t in tasks])
 
     monkeypatch.setattr(
-        'cylc.flow.flags.cylc7_back_compat',
-        True)
+        'cylc.flow.flags.cylc7_back_compat', True)
     monkeypatch.setattr(
         'cylc.flow.task_events_mgr.TaskEventsManager._insert_task_job',
         lambda *_: True)
 
     schd = scheduler(flow(config))
     async with start(schd):
-        for task_status, new_runahead in states:
```

Comment:
This was better done with a `for` loop than with `@pytest.mark.parametrize`.

This is now 8 separate tests, each of which starts a new scheduler taking 8 times as long to run.

Reply:
Hmm, right. I think I did that to make test failure diagnosis easier during development (when some were failing). I'll revert it ...

Reply:
Admittedly, failure diagnostics is a little easier with params, although leaving a `print` statement in the `for` loop has similar results.

The way to param efficiently is to break the common part into a module-scoped fixture. Due to the way we've set up Pytest, all tests in a module will be run in the same test runner even when distributing tests with pytest-xdist so a module-scoped fixture will only get created once per test session:

```python
@pytest.fixture(scope='module')
async def mod_workflow(mod_flow, mod_scheduler, mod_start):
    # set up a workflow and put it in the desired state
    id_ = mod_flow(...)
    schd = mod_scheduler(flow)
    async with mod_start(schd):
        yield schd

@pytest.mark.parametrize(...)
def test_foo(mod_workflow, input, output):
    # this is fine so long as some_function does not modify the scheduler state
    assert some_function(input) == output
```

The Cylc Pytest fixtures have module-scoped variants prefixed with `mod_` which do the same thing as their non-prefixed counterparts.

Reply:
Done, went the module-scoped way, with params.